src/messaging/messages/response-message.js

```diff
@@ -12,9 +13,11 @@ export default class ResponseMessage extends Message {
    */
   constructor (request, body = {}, responseCode = ResponseMessage.responseCodes.UNDEFINED) {
     super(body)
+    if (!request) throw new Error('Request is not provided')
+    if (!request.ID) throw new Error('Request has no ID')
     this.role = Message.roles.RESPONSE
     this.requestHeader = request.header || {}
-    this.requestHeader.ID = request.ID // ID of the request to match request and response
+    this.requestID = request.ID // ID of the request to match request and response
```

Comment:
Updated to not accidentally overwrite an `ID` prop inside a header, if it will be there